tests/jballerina-unit-test/src/test/resources/test-src/record/readonly_record_fields.bal

```diff
@@ -924,6 +924,61 @@ function assertTrue(any|error actual) {
     assertEquality(true, actual);
 }
 
+type RecordWithNeverField record {|
+    never x?;
+|};
+
+type OpenRecordWithNever record {
+    never x?;
+};
+
+type RecordWithNeverAndReadonly record {|
+    never x?;
+    readonly int y;
+|};
+
+type RecordWithNeverWithoutReadonly record {|
+    never x?;
+    string y;
+|};
+
+type RecordWithNeverFieldRecord record {|
+       record {|
+           never a;
+       |} x?;
+   |};
+
+type RecordWithNeverUnion record {|
+    never|never x?;
+|};
+
+function testNeverFieldRecord() {
+    record {|
+        never x?;
+        never y?;
+    |} c = {};
+
+    readonly d = c;
+    RecordWithNeverField e = {};
+    readonly e1 = e;
+    OpenRecordWithNever f = {};
+    RecordWithNeverAndReadonly g = {y: 1};
+    readonly g1 = g;
+    RecordWithNeverWithoutReadonly h = {y: "abc"};
+    RecordWithNeverFieldRecord i = {};
+    readonly i1 = i;
+    RecordWithNeverUnion j = {};
+    readonly j1 = j;
+
+    assertTrue(d is record {|never x?; never y?;|} & readonly);
+    assertTrue(e1 is record {|never x?;|} & readonly);
+    assertTrue(f is record {|never x?; anydata...;|});
```

Comment:
To make sure f is not readonly. 
```suggestion
    assertFalse(f is record {|never x?; anydata...;|} & readonly);
```

Otherwise,  defining `record { never x?; } f` and checking`f is record {| never x?; anydata...; |}` is a obviously known `true`.